google/cloud/storage/client.py

```diff
@@ -798,17 +809,28 @@ def _bucket_arg_to_bucket(self, bucket_or_name):
                  str, \
             ]):
                 The bucket resource to pass or name to create.
+            generation (Optional[int]):
+                The bucket generation. Only buckets with soft delete policies
+                have generations. If generation is specified, bucket_or_name
+                must be a name (str).
 
```

Comment:
Let's remove this line too "Only buckets with soft delete policies have generations."